scripts/build-catalogue.in

```diff
@@ -74,32 +87,66 @@ name    = re.sub(r'_+', r'_',
                         args['name']))
 
 mod_dir = pwd / Path(args['modpfx'])
-mods    = [ f[:-4] for f in os.listdir(mod_dir) if f.endswith('.mod') ]
-verbose = args['verbose'] and not args['quiet']
+mods    = [f[:-4] for f in os.listdir(mod_dir) if f.endswith('.mod')]
 quiet   = args['quiet']
+verbose = args['verbose'] and not quiet
 debug   = args['debug']
+raw   = args['raw']
+gpu     = args['gpu']
+
+if gpu:
+    if gpu == 'cuda':
+        gpu_support = """add_compile_definitions(ARB_CUDA)
+add_compile_definitions(ARB_HAVE_GPU)
+
+enable_language(CUDA)
+set(CMAKE_CUDA_HOST_COMPILER @CMAKE_CXX_COMPILER@)
+set(CMAKE_CUDA_ARCHITECTURES @CMAKE_CUDA_ARCHITECTURES@)"""
+    else:
+        print(f"Unsupported GPU target: {gpu}. If you need support for HIP or Clang-CUDA, please check here: https://github.com/arbor-sim/arbor/issues/1783")
+        exit(-1)
+else:
+    gpu_support = "# GPU: Disabled"
+
+this_path = Path(__file__).parent
+data_path = (this_path / "@ARB_REL_DATADIR@").resolve()
+pack_path = (this_path / "@ARB_REL_PACKAGEDIR@").resolve()
+exec_path = this_path.resolve()
```

Comment:
This change to relative paths is orthogonal, but related. Relative paths make this script more robust against being shuffled around, so in a sense it _enables_ sk-build rather than depending on it.